javaCreateAccountCore/src/main/java/com/hamdan/forzenbook/java/createaccount/core/viewmodel/JavaCreateAccountState.java

```diff
@@ -0,0 +1,83 @@
+package com.hamdan.forzenbook.java.createaccount.core.viewmodel;
+
+import androidx.annotation.Nullable;
+
+import com.hamdan.forzenbook.java.core.Entry;
+import com.hamdan.forzenbook.java.core.UserEntryError;
+
+public class JavaCreateAccountState {
+
+    @Nullable
+    final private Integer errorId;
+    final private Entry firstName;
+    final private Entry lastName;
+    final private Entry birthDay;
+    final private Entry email;
+    final private Entry location;
+    final private Boolean isDateDialogOpen;
+    final private Boolean isLoading;
+
+    public JavaCreateAccountState(
+            @Nullable Integer id,
+            Entry first,
+            Entry last,
+            Entry date,
+            Entry inEmail,
+            Entry inLocation,
+            Boolean dialogState,
+            Boolean loadingState
+    ) {
+        errorId = id;
+        firstName = first;
+        lastName = last;
+        birthDay = date;
+        location = inLocation;
+        email = inEmail;
+        isDateDialogOpen = dialogState;
+        isLoading = loadingState;
+    }
+
+    public JavaCreateAccountState() {
+        errorId = null;
+        firstName = new Entry("", new UserEntryError());
+        lastName = new Entry("", new UserEntryError());
+        birthDay = new Entry("", new UserEntryError());
+        location = new Entry("", new UserEntryError());
+        email = new Entry("", new UserEntryError());
+        isDateDialogOpen = false;
+        isLoading = false;
+    }
+
+    @Nullable
```

Comment:
If you're going to annotate one function annotate them all

Reply:
annotated